libbeat/processors/add_cloud_metadata/http_fetcher.go

```diff
@@ -129,16 +130,21 @@ func (f *httpMetadataFetcher) fetchRaw(
 func getMetadataURLs(c *common.Config, defaultHost string, metadataURIs []string) ([]string, error) {
 	var urls []string
 	config := struct {
-		MetadataHostAndPort string `config:"host"` // Specifies the host and port of the metadata service (for testing purposes only).
+		MetadataHostAndPort string            `config:"host"` // Specifies the host and port of the metadata service (for testing purposes only).
+		TLSConfig           *tlscommon.Config `config:"ssl"`
 	}{
 		MetadataHostAndPort: defaultHost,
 	}
 	err := c.Unpack(&config)
 	if err != nil {
 		return urls, errors.Wrap(err, "failed to unpack add_cloud_metadata config")
 	}
+	scheme := "http"
+	if config.TLSConfig.IsEnabled() {
+		scheme = "https"
+	}
 	for _, uri := range metadataURIs {
-		urls = append(urls, "http://"+config.MetadataHostAndPort+uri)
+		urls = append(urls, scheme+"://"+config.MetadataHostAndPort+uri)
```

Comment:
By changing the scheme for all providers to HTTPS it means that the processor will really only ever work for openstack-nova when TLS is configured. So if you were deploying in a multiple environments you would need to customize the config for each one, minimizing the effectiveness of the auto-detection.

Perhaps we could register two instances of the openstack-nova provider, one for HTTP and one for HTTPS. `provider` would get an extra flag to indicate https. Then the `add_cloud_metadata` would automatically work any provider (including openstack-nova w/ https enabled if the machine trusted the CA cert). The user's TLS config options would used with any provider connecting over TLS.

Reply:
Ok, I have added it as a different provider called `openstack-ssl`/`nova-ssl`. Let me know what you think.